src/pint/models/fdjump.py

```diff
@@ -0,0 +1,207 @@
+                description="Whether to use log-frequency for computing FDJUMPs.",
+            )
+        )
+        for j in range(1, fdjump_max_index + 1):
+            self.add_param(
+                maskParameter(
+                    name=f"FD{j}JUMP",
+                    units="second",
+                    description=f"System-dependent FD parameter of polynomial index {j}",
+                )
+            )
+
+        self.delay_funcs_component += [self.fdjump_delay]
+
+    def setup(self):
+        super().setup()
+
+        self.fdjumps = [
+            mask_par
+            for mask_par in self.get_params_of_type("maskParameter")
+            if self.param_regex.match(mask_par)
+        ]
+
+        for fdj in self.fdjumps:
+            # prevents duplicates from being added to phase_deriv_funcs
+            if fdj in self.deriv_funcs.keys():
+                del self.deriv_funcs[fdj]
+            self.register_deriv_funcs(self.d_delay_d_FDJUMP, fdj)
+
+    def get_fd_index(self, par):
+        """Extract the FD index from an FDJUMP parameter name. In a parameter name
+        "FDpJUMPq", p is the FD/prefix index and q is the mask index.
+
+        Parameters
+        ----------
+        par: Parameter name (str)
+
+        Returns
+        -------
+        FD index (int)
+        """
+        if m := self.param_regex.match(par):
+            return int(m.groups()[0])
+        else:
+            raise ValueError(
+                f"The given parameter {par} does not correspond to an FDJUMP."
+            )
+
+    def get_freq_y(self, toas):
+        """Get frequency or log-frequency in GHz based on the FDJUMPLOG value.
+        Returns (freq/1_GHz) if FDJUMPLOG==N and log(freq/1_GHz) if FDJUMPLOG==Y.
+        Any non-finite values are replaced by zero.
+
+        Parameters
+        ----------
+        toas: pint.toa.TOAs
+
+        Returns
+        -------
+        (freq/1_GHz) or log(freq/1_GHz) depending on the value of FDJUMPLOG (float).
+        """
+        tbl = toas.table
+        try:
+            freq = self._parent.barycentric_radio_freq(toas)
+        except AttributeError:
+            warn("Using topocentric frequency for frequency dependent delay!")
+            freq = tbl["freq"]
+
+        y = (
+            np.log(freq.to(u.GHz).value)
+            if self.FDJUMPLOG.value
+            else freq.to(u.GHz).value
+        )
+        non_finite = np.invert(np.isfinite(y))
+        y[non_finite] = 0.0
+
+        return y
+
+    def fdjump_delay(self, toas, acc_delay=None):
+        """Calculate frequency dependent delay.
+
+        If FDJUMPLOG is Y, use the following expression:
+            Z. Arzoumanian, The NANOGrav Nine-year Data Set: Observations, Arrival
+            Time Measurements, and Analysis of 37 Millisecond Pulsars, The
+            Astrophysical Journal, Volume 813, Issue 1, article id. 65, 31 pp.(2015).
+            Eq.(2):
+            FDJUMP_delay = sum_i(c_i * (log(obs_freq/1GHz))^i)
```

Comment:
add better line breaks to make this more clear in RTD (https://nanograv-pint--1615.org.readthedocs.build/en/1615/_autosummary/pint.models.fdjump.FDJump.html#pint.models.fdjump.FDJump.fdjump_delay).  You could also use a proper math rendering (although that is harder to read in code)

Reply:
I have removed the reference as it looks cluttered. That paper does not discuss FDJUMPs anyway.
I have instead mentioned that the log-expression is similar to global FD 

I think proper math rendering will make it harder to read in code. So I am keeping the expressions as they are

Reply:
Added line breaks